### PR TITLE
feat: implement query profile

### PR DESCRIPTION
**Pull request by kolea2**
Depends on merging https://github.com/googleapis/java-datastore/pull/1278

### CODE REVIEW

**Review by bhshkh — APPROVED** [review]

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResultsImpl.java [hunk 1]
```diff
@@ -86,9 +94,14 @@ private void sendRequest() {
       // projection entity can represent all type of results
       actualResultType = ResultType.PROJECTION_ENTITY;
     }
+    boolean isQueryPlan = actualResultType.getQueryType() == null && queryMode == QueryMode.PLAN;
     Preconditions.checkState(
-        queryResultType.isAssignableFrom(actualResultType),
+        queryResultType.isAssignableFrom(actualResultType) || isQueryPlan,
         "Unexpected result type " + actualResultType + " vs " + queryResultType);
```
Comment (cindy-peng) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResultsImpl.java:100]: Nitpick: Error message can also include scenario when it is not a query plan?

**Review by cindy-peng — APPROVED** [review]

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResultsImpl.java (hunk 1 above)
Comment (kolea2) [reply google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResultsImpl.java]: fixed, thanks!

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java [hunk 2]
```diff
@@ -462,6 +464,29 @@ interface TransactionCallable<T> {
...
+   * QueryResults<Entity> results = datastore.run(query, QueryMode.EXPLAIN_ANALYZE);
+   * ResultSetStats resultSetStats = results.getResultSetStats();
+   * }</pre>
+   *
+   * @throws DatastoreException upon failure
+   */
+  @BetaApi
+  default <T> QueryResults<T> run(
+      Query<T> query, QueryProfile.QueryMode queryMode, ReadOption... options) {
+    throw new UnsupportedOperationException("not implemented");
```
[the remark this answers is not in the document]
Comment (kolea2) [reply google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java]: fixed, thanks!

[the remark this answers is not in the document]
Comment (kolea2) [reply google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java]: fixed, thanks!

**Pull request closed without merging**